node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -312,7 +312,10 @@ case class DataMessageHandler(
     logger.info(s"Received inv=${invMsg}")
     val getData = GetDataMessage(invMsg.inventories.map {
       case Inventory(TypeIdentifier.MsgBlock, hash) =>
-        Inventory(TypeIdentifier.MsgFilteredBlock, hash)
+        // only request the merkle block if we are spv enabled
+        if (appConfig.isSPVEnabled) {
+          Inventory(TypeIdentifier.MsgFilteredBlock, hash)
+        } else Inventory(TypeIdentifier.MsgBlock, hash)
```

Comment:
Hmm.... so is this going to cause us to fetch a full block twice when we are running in neutrino mode?

Reply:
No, the inventory will just send us a list of blocks, not the full blocks, so after receiving it this way, we can then request the full one.

https://developer.bitcoin.org/reference/p2p_networking.html#inv

Reply:
My main concern here is that things are working _without_ sending this `TypeIdentifier.MsgBlock` back to the peer in neutrino mode. So do we really need to send it again? If we are talking about #1093 this `TypeIdentifier.MsgBlock` doesn't seem necessary since neutrino works as expected :tm: but we are just receiving unecessary spv messages

Reply:
According to [this](https://developer.bitcoin.org/reference/p2p_networking.html#inv), inventory messages will be sent for "announcements of new transactions or blocks, or it can be sent in reply to a “getblocks” message or “mempool” message."

In the case we are in neutrino mode we could be sending `getblocks` for filter matches and in those cases we would want to process the block, not the filtered block. It may be we have just gotten lucky we haven't run into the issue yet.